### PR TITLE
Increase test coverage, remove unnecessary parts, deploy to goerli

### PR DESCRIPTION
**Pull request by bxmmm1**
Minor gauge redirect refactor, we don't need a state variable, we can call it when we create a pool. Also I've removed unused variable, added vote delegate test

### CODE REVIEW

**Comment by bxmmm1** [comment]
@BritikovKI, @arseneeth  I've made some changes, and I need a review.

At the moment in our master branch we have the following execution:

When want to get rewards we call `earmarkRewards()` in the controller, and it calls `stash.claimRewards()`

claimRewards on first run calls redirect `IController(operator).setGaugeRedirect(pid);`, and that state `hasRedirected` is flipped to true, meaning that gaugeRedirect runs only once.

`setGaugeRedirect(pid)` is calling the controller, and the controller is calling voterProxy to redirect gauge rewards to that stash.

so flow is:
Controller.earmarkRewards -> stash.ClaimRewards() -> IF NOT REDIRECTED controller.setGaugeRedirect() -> voterProxy.execute() -> gauge.set_rewards_receiver(stashAddress)

if we redirected it already, we waste gas on cold SLOAD (2100 gas per tx)

What I've done is changed it to the following:
controller.addPool() creates a stash for that pool, and after the creation of the stash we immediately call voterProxy to redirect rewards to that stash for that gauge.

This way we've removed an extra state variable, and load from storage each time we `claimRewards`, thus making our tx cheaper (SLOAD from cold storage = 2100 GAS)

So instead of checking if we redirected gauge every time we claim rewards, we redirect gauge after the creation of stash on addPool.

Please comment if you see something weird, or something that doesn't make sense.

File: contracts/Controller.sol [hunk 1]
```diff
@@ -241,6 +241,7 @@ contract Controller is IController {
         poolInfo[pid].stash = stash;
         IVoterProxy(staker).grantStashAccess(stash);
         IRewardFactory(rewardFactory).grantRewardStashAccess(stash);
+        redirectGaugeRewards(pid);
```
Comment (BritikovKI) [inline contracts/Controller.sol:244]: Why do we need it?

File: contracts/Controller.sol [hunk 2]
```diff
@@ -217,7 +217,7 @@ contract Controller is IController {
         //create a reward contract for bal rewards
         address newRewardPool = IRewardFactory(rewardFactory).createBalRewards(pid, token);
         //create a stash to handle extra incentives
-        address stash = IStashFactory(stashFactory).createStash(pid, _gauge, staker);
```
Comment (BritikovKI) [inline contracts/Controller.sol:220]: Why staker is removed?

File: deploy/01_d2dBal_and_balDepositor.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ const deployFunction = async ({ getNamedAccounts, deployments }) => {
   });
 
   // Owner of D2DBal should be BalDepositor
-  await execute('D2DBal', { from: root, log: true }, 'transferOwnership', balDepositor)
+  await execute('D2DBal', { from: root, log: true, gasLimit: 6000000 }, 'transferOwnership', balDepositor)
```
Comment (BritikovKI) [inline deploy/01_d2dBal_and_balDepositor.js:23]: Why do we need to set the gasLimit strictly?

File: test/unit/controller.js [hunk 4]
```diff
@@ -259,9 +259,6 @@ describe('unit - Controller', function () {
 
     const { stash } = await controller.poolInfo(0);
 
-    // revert if unauthorized
-    await expect(controller.setGaugeRedirect(0)).to.be.revertedWith('Unauthorized()');
```
Comment (BritikovKI) [inline test/unit/controller.js:263]: Why not anymore revert?

**Review by BritikovKI — COMMENTED** [review]
Overall it's good tbh

File: contracts/Controller.sol [hunk 5]
```diff
@@ -484,4 +473,13 @@ contract Controller is IController {
         feeToken.transfer(lockFees, _balance);
         IRewards(lockFees).queueNewRewards(_balance);
     }
+
+    /// @notice redirects rewards from gauge to rewards contract
+    /// @param _pid the id of the pool
+    function redirectGaugeRewards(uint256 _pid) private {
```
Comment (BritikovKI) [inline contracts/Controller.sol:479]: Is it possible that we change the stash? If it is, maybe make it public?

File: contracts/Controller.sol (hunk 2 above)
Comment (bxmmm1) [reply contracts/Controller.sol:220]: Unused in factory

File: deploy/01_d2dBal_and_balDepositor.js (hunk 3 above)
Comment (bxmmm1) [reply deploy/01_d2dBal_and_balDepositor.js:23]: When there is a big network load, it fails to determine gasLimit automatically, and doesn't execute tx, throws an error

File: contracts/Controller.sol (hunk 5 above)
Comment (bxmmm1) [reply contracts/Controller.sol]: it is created in `addPool` via factory, and can't be changed

**Review by BritikovKI — APPROVED** [review]
Lgtmt

**Pull request merged**